app/controllers/sessions_controller.rb

```diff
@@ -175,4 +182,9 @@ def session_active?
   def login_conditions_met?
     @user&.mfa_enabled? && @user&.otp_verified?(params[:otp]) && session_active?
   end
+
+  def record_mfa_login_duration(mfa_type:)
+    duration = Time.now.utc - session[:mfa_created_at]
```

Comment:
I need to think more on this. Is it possible for `session[:mfa_created_at]` to be `nil`? When it is, how do we want to handle that.

Reply:
> Is it possible for `session[:mfa_created_at]` to be `nil`?

We encountered an [error](https://app.honeybadger.io/projects/40972/faults/92632858) from `session[:mfa_expires_at]` being `nil` during/right after deploy. This is because users could be on the MFA page when the change is deployed and in those cases, when someone MFAs, `session[:mfa_expires_at]` won't be set and would throw an error.

I saw you added a change to return when nil, I think that's the best way to handle it 👍